### PR TITLE
feat(taskframework): support modify job parameters

### PR DESCRIPTION
**Pull request by krihy**
#### What type of PR is this?

type-feature
module-taskframework
#### What this PR does / why we need it:
job parameters can be modified when job is not terminated

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/caller/BaseJobCaller.java [hunk 1]
```diff
@@ -142,6 +144,51 @@ protected void afterStopFailed(JobIdentity ji, Exception e) throws JobException
...
+            throw new JobException("Modify job parameters not succeed, jobId={0}, response={1}", ji.getId(),
+                    JsonUtils.toJson(e.getMessage()));
+        }
+    }
+
+    private String getExecutorPoint(JobIdentity ji, JobConfiguration jobConfiguration, JobEntity jobEntity)
+            throws JobException {
+        String executorEndpoint = jobEntity.getExecutorEndpoint();
+        if (executorEndpoint == null) {
+            try {
```
Comment (guowl3) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/caller/BaseJobCaller.java:176]: I don't think retrying is necessary

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/schedule/JobScheduler.java [hunk 2]
```diff
@@ -45,15 +47,24 @@ public interface JobScheduler {
...
     void cancelJob(Long jobId) throws JobException;
 
+    /**
+     * modify job parameters, new parameters will be delivery to executor if job is running
+     *
+     * @param jobId job id
+     * @param jobParameters job parameters
+     * @throws JobException throw JobException if job is terminated
+     */
+    void modifyJobParameters(Long jobId, Consumer<Map<String, String>> jobParameters) throws JobException;
```
Comment (guowl3) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/schedule/JobScheduler.java:57]: The input parameters of the method are a bit complicated.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/util/JobUtils.java [hunk 3]
```diff
@@ -69,6 +72,22 @@ public static <T> T fromJson(String json, Class<T> clazz) {
         return new Gson().fromJson(json, clazz);
     }
 
+    public static <T> T fromJson(String json, Type type) {
```
Comment (guowl3) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/util/JobUtils.java:75]: Is there a better solution?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/caller/BaseJobCaller.java (hunk 1 above)
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/caller/BaseJobCaller.java]: okey, done

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/schedule/JobScheduler.java (hunk 2 above)
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/schedule/JobScheduler.java]: modify method parameter, remove consumer

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/util/JobUtils.java (hunk 3 above)
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/util/JobUtils.java:75]: this is solution is more easier

**Review by guowl3 — APPROVED** [review]
LGTM

**Pull request merged**